### PR TITLE
flipper-bids-pass-rough-spec.k passing

### PR DESCRIPTION
**Pull request by goodlyrottenapple**
Due to https://github.com/runtimeverification/haskell-backend/issues/3060, the definition and simplification rule for `#WordPackAddrUInt48UInt48` was tweaked slightly to allow the proof in `flipper-bids-pass-rough-spec.k` to pass.

### CODE REVIEW

File: tests/specs/mcd/word-pack.k [hunk 1]
```diff
@@ -78,6 +87,9 @@ module WORD-PACK-COMMON
     syntax Int ::= #WordPackUInt48UInt48     (       Int , Int ) [function, no-evaluators, smtlib(WordPackUInt48UInt48)]
                  | #WordPackAddrUInt48UInt48 ( Int , Int , Int ) [function, no-evaluators, smtlib(WordPackAddrUInt48UInt48)]
                  | #WordPackAddrUInt8        (       Int , Int ) [function, no-evaluators, smtlib(WordPackAddrUInt8)]
+
+    syntax Bool ::= #WordPackAddrUInt48UInt48Prop ( Int , Int , Int, Int ) [function, smtlib(WordPackAddrUInt48UInt48Bool)]
+
  // -----------------------------------------------------------------------------------------------------------------
```
Comment (ehildenb) [inline tests/specs/mcd/word-pack.k:93]:
```suggestion
                 | #WordPackAddrUInt8        (       Int , Int ) [function, no-evaluators, smtlib(WordPackAddrUInt8)]
 // -----------------------------------------------------------------------------------------------------------------
 
    syntax Bool ::= #WordPackAddrUInt48UInt48Prop ( Int , Int , Int, Int ) [function, smtlib(WordPackAddrUInt48UInt48Bool)]
 // -----------------------------------------------------------------------------------------------------------------------
```

File: tests/specs/mcd/word-pack.k [hunk 2]
```diff
@@ -53,6 +53,15 @@ module WORD-PACK-HASKELL [kore]
...
 
+
+    rule     #WordPackAddrUInt48UInt48Prop(ADDR_UINT48_UINT48, ADDR, UINT48_1, UINT48_2)
+          => true
+     ensures ADDR     ==Int maxUInt160 &Int  ADDR_UINT48_UINT48
+     andBool UINT48_1 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow160)
+     andBool UINT48_2 ==Int maxUInt48  &Int (ADDR_UINT48_UINT48 /Int pow208)
+     andBool #rangeUInt(256, ADDR_UINT48_UINT48)
+       [simplification]
+
```
Comment (ehildenb) [inline tests/specs/mcd/word-pack.k:64]: Can you transform all the other uses of `#WordPack...` to this way as well?

**Comment by ehildenb** [comment]
You likely will have to disable this proof on the java backend (via `tests/failing-symbolic.java`), because the new abstraction only works for Haskell backend.

**Comment by goodlyrottenapple** [comment]
Should we keep both versions of `#WordPack...` and `#WordPack...Prop` and have `ADDR ==Int #WordPack... andBool #WordPack...Prop(ADDR, ...)` in all the claims that mention it?

**Comment by ehildenb** [comment]
Does the spec pass locally (on both backends, with `make tests/specs/mcd/flipper-bit-pass-rough-spec.k.prove TEST_SYMBOLIC_BACKEND=java` and `... TEST_SYMBOLIC_BACKEND=haskell`) if you do this? If so, then yes, let's go with that solution for now.

**Comment by ehildenb** [comment]
Also, you should make the same adjustments to all the occurances of `WordPack*` (there are 3 types of `WordPack*`), and see if any more proofs pass because of that.

**Pull request closed without merging**